Homework27/src/Files/Task1.java

```diff
@@ -0,0 +1,103 @@
+package Files;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.concurrent.LinkedBlockingQueue;
+
+public class Task1 {
+
+	private static LinkedBlockingQueue<Runnable> queue;
+	private static ArrayList<Worker> workers;
+	private static BufferedReader br = null;
+	private static Integer counter = 0;
+	private static char test;
+
+	public static void main(String[] args) {
+		queue = new LinkedBlockingQueue<Runnable>();
+
+		try {
+			br = new BufferedReader(new FileReader(new File("text.txt")));
+
+			String firstLine = br.readLine();
+			System.out.println(firstLine);
+			test = firstLine.charAt(0);
+			queue.add(new Task(firstLine, test));
+
+			while (br.ready()) {
+
+				queue.add(new Task(br.readLine(), test));
+			}
+			workers = new ArrayList<Worker>();
+			for (int i = 0; i < 4; i++) {
+				Worker worker = new Worker();
+				worker.start();
+				try {
+					worker.join();
+
+				} catch (InterruptedException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
+				}
+				workers.add(worker);
+			}
+
+		} catch (FileNotFoundException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		System.out.println("There is " + counter + " letters " + test
+				+ " that repeats in file.");
+	}
+
+	public static class Worker extends Thread {
+		@Override
+		public void run() {
```

Comment:
Ako u ovu metodu ubacim counter da vidim koliko je koji Worker procesirao job-ova u queue dobijem:
Worker 0 done 1135 jobs.
Worker 1 done 0 jobs.
Worker 2 done 0 jobs.
Worker 3 done 0 jobs.

Sto znaci da je prvi worker uradio sve jobove, dok ostali nisu nista. Zasto je to tako? :)